MapboxCoreNavigation/RouteController.swift

```diff
@@ -157,6 +192,53 @@ open class RouteController: NSObject {
         locationManager.stopUpdatingLocation()
         locationManager.stopUpdatingHeading()
     }
+    
+    /**
+     Send feedback about the current road segment/maneuver to the Mapbox data team.
+     
+     You can hook this up to a custom feedback UI in your app to flag problems during navigation
+     such as road closures, incorrect instructions, etc. 
+     
+     With the help of a custom `description` to elaborate on the nature of the problem, using
+     this function will automatically flag the road segment/maneuver the user is currently on for 
+     closer inspection by Mapbox's system and team.
+     */
+    public func sendFeedback(type: FeedbackType, description: String?) {
+        enqueueFeedbackEvent(type: type, description: description)
+    }
+}
+
+extension RouteController {
+    func progressDidChange(notification: NSNotification) {
+        if sessionState.departureTimestamp == nil {
+            sessionState.departureTimestamp = Date()
+            sendDepartEvent()
+        }
+        checkAndSendOutstandingFeedbackEvents()
+    }
+    
+    func alertLevelDidChange(notification: NSNotification) {
+        let alertLevel = routeProgress.currentLegProgress.alertUserLevel
+        if alertLevel == .arrive && sessionState.arrivalTimestamp == nil {
+            sessionState.arrivalTimestamp = Date()
+            sendArriveEvent()
+        }
+    }
+    
+    func willReroute(notification: NSNotification) {
+        enqueueRerouteEvent()
+    }
+    
+    func didReroute(notification: NSNotification) {
+        let route = routeProgress.route
+        if let lastReroute = outstandingFeedbackEvents.filter({$0 is RerouteEvent }).last {
+            if let geometry = route.coordinates {
+                lastReroute.eventDictionary["newGeometry"] = Polyline(coordinates: geometry).encodedPolyline
+                lastReroute.eventDictionary["newDistanceRemaining"] = round(route.distance)
+                lastReroute.eventDictionary["newDurationRemaining"] = round(route.expectedTravelTime)
```

Comment:
Would it be possible to add some tests for at least the event generation? The spec may change a lot still so relying on tests may actually make more sense at this point (or always).